day03/create_ec2.sh

```
@@ -30,62 +29,63 @@ wait_for_instance() {
+	SUBNET_ID="subnet-01845bee6f9f54abb "
+	SECURITY_GROUP_IDS="sg-0ddba97c267e46e49"  # Add your security group IDs separated by space
+	INSTANCE_NAME="Shell-Script-EC2-Demo"
+
+	#call the function to create ec2
+	create_ec2_instance "$AMI_ID" "$INSTANCE_TYPE" "$KEY_NAME" "$SUBNET_ID" "$SECURITY_GROUP_IDS" "$INSTANCE_NAME"
+	echo " EC2 instance creation completed"
+}
 
-    # Call the function to create the EC2 instance
-    create_ec2_instance "$AMI_ID" "$INSTANCE_TYPE" "$KEY_NAME" "$SUBNET_ID" "$SECURITY_GROUP_IDS" "$INSTANCE_NAME"
+main"$@"
```

Shellcheck (0.11.0)

[error] Argument mixes string and array. Use * or separate argument.

(SC2145)